src/abort.rs

```diff
@@ -64,3 +59,35 @@ impl<I, V, E> IteratorExt<I, V, E> for I
     }
 }
 
+impl<I, V, IE, OE> IteratorExt<I, V, IE> for Result<I, OE>
+    where I: Iterator<Item = Result<V, IE>> + Sized,
+          OE: Debug
+{
+    fn abort_on_err(self) -> AbortingIter<I, V, IE> {
+        AbortingIter::from(self.unwrap_or_abort())
+    }
+}
+
+
+/// Extension trait for convenient abortion in case of errors
+///
+pub trait Abortable<V>
+{
+    /// Just like a regulat unwrap() except it performs proper logging
+    ///
+    /// Returns the contained value or aborts the program, logging the error.
+    ///
+    fn unwrap_or_abort(self) -> V;
+}
+
+impl<V, E> Abortable<V> for Result<V, E>
+    where E: Debug
```

Comment:
I'm not that sure here, but as we're using `error_chain` I'd rather use the actual error type and `error_chain`s tracing functionality and not the `error!(...)` logging macro...